src/components/SelectionPills/SelectionPill/SelectionPill.js

```diff
@@ -0,0 +1,150 @@
+     * @param {props} object All the props passed to the component
+     */
+    onFocus: PropTypes.func,
+
+    /** Callback function called after pill has lost focus
+     * @param {SyntheticEvent} event The react `SyntheticEvent`
+     * @param {props} object All the props passed to the component
+     */
+    onBlur: PropTypes.func,
+
+    /** Snacks theme attributes provided by Themer */
+    snacksTheme: themePropTypes,
+
+    /** Text to appear inside pill */
+    text: PropTypes.string,
+
+    /** Optional style overrides for button and its states */
+    style: PropTypes.shape({
+      button: PropTypes.object,
+      disabledStyle: PropTypes.object,
+      selectedStyle: PropTypes.object,
+      focusedStyle: PropTypes.object,
+    }),
+
+    /** Aria overrides for accessibility (i.e. use if label is not descriptive enough for screen readers) */
+    aria: PropTypes.shape({
+      label: PropTypes.string,
+    }),
+  }
+
+  static defaultProps = {
+    elementAttributes: {},
+    isSelected: false,
+    parentControlledState: false,
+    onClick: NoOp,
+    onFocus: NoOp,
+    onBlur: NoOp,
+    style: {},
+    aria: {},
+  }
+
+  state = {
+    isSelected: this.props.isSelected,
+    isFocused: false,
+  }
+
+  handleChange = event => {
+    const { onClick } = this.props
+    const { isSelected } = this.state
+
+    this.setState({ isSelected: !isSelected })
+    onClick(event, { ...this.props, isSelected: !isSelected })
+  }
+
+  toggleFocus = () => {
+    const { isFocused } = this.state
+    this.setState({ isFocused: !isFocused })
+  }
+
+  handleFocus = event => {
+    this.toggleFocus()
+    this.props.onFocus(event, { ...this.props, isFocused: !this.state.isFocused })
+  }
+
+  handleBlur = event => {
+    this.toggleFocus()
+    this.props.onBlur(event, { ...this.props, isFocused: !this.state.isFocused })
+  }
+
+  renderInputBtn({ selected }) {
+    const { aria, isDisabled, id } = this.props
+    const componentStyles = getStyles({})
+
+    return (
+      <div style={componentStyles.checkBoxOverrideStyle}>
+        <input
+          id={id}
+          type="checkbox"
+          onChange={this.handleChange}
+          checked={selected}
+          disabled={isDisabled}
+          aria-label={aria.label}
+          onBlur={this.handleBlur}
+          onFocus={this.handleFocus}
+        />
+      </div>
+    )
+  }
+
+  render() {
+    const { id, snacksTheme, text, style, isDisabled, parentControlledState } = this.props
+    const { isFocused } = this.state
+    const { primaryForeground } = snacksTheme.colors
+    const selected = parentControlledState ? this.props.isSelected : this.state.isSelected
+    const componentStyles = getStyles({
+      isSelected: selected,
+      isFocused,
+      isDisabled,
+      primaryForeground,
+      externalStyles: style,
+    })
+
+    return (
+      <li style={componentStyles.listElement} {...this.props.listElementAttributes}>
```

Comment:
Let me know if `listElementAttributes` was for a specific reason, but this could be a good time to start the practice of spreading extra props to components without a separate prop like `elementAttributes`. Thinking something like 
```
const { id, snacksTheme, text, style, isDisabled, parentControlledState, ...other } = this.props

return (
      <li style={componentStyles.listElement} {...other}>
      ...
)
```

Reply:
Definitely no good reason for this. More following practice seen throughout snacks, but very supportive of moving towards spreading other props. Will update 👍

Reply:
Actually, doing this spreads a bunch of inappropriate objects on the list. In order to get around this, I would need to explicitly deconstruct all possible props even if they were not used here. It doesn't feel like the right treatment in this case.